### PR TITLE
Show all students (and only students) in assignments dashboards

### PR DESCRIPTION
**Pull request by marcospri**
Join data from H for the dashboard with LMS DB records
Use the LMS DB data to:

- Filter out teachers
- Include students that didn't make an annotation

## Testing

I've been applying a hacky diff like:

```diff
diff --git a/lms/views/dashboard.py b/lms/views/dashboard.py
index b3388208a..fa250873d 100644
--- a/lms/views/dashboard.py
+++ b/lms/views/dashboard.py
@@ -51,7 +51,6 @@ class DashboardViews:
         route_name="api.assignment.stats",
         request_method="GET",
         renderer="json",
-        permission=Permissions.DASHBOARD_VIEW,
     )
     def api_assignment_stats(self):
         """Fetch the stats for one particular assignment."""
@@ -96,6 +95,7 @@ class DashboardViews:
 
     def get_request_assignment(self):
         assignment = self.assignment_service.get_by_id(self.request.matchdict["id_"])
+        return assignment
         if not assignment:
             raise HTTPNotFound()
```

Finding a relevant assignment ID in the DB and accessing the endpoint via: http://localhost:8001/api/assignment/292/stats

### CODE REVIEW

File: lms/models/assignment.py [hunk 1]
```diff
@@ -81,6 +81,10 @@ class Assignment(CreatedUpdatedMixin, Base):
     groupings = association_proxy("assignment_grouping", "grouping")
     """List of groupings this assigments is related to."""
 
+    membership = sa.orm.relationship(
```
Comment (marcospri) [inline lms/models/assignment.py:84]: Add new field to the assignments to get all "users" without having to write the query each time.

File: lms/services/assignment.py [hunk 2]
```diff
@@ -198,11 +200,19 @@ def get_by_id(self, id_: int) -> Assignment | None:
 
     def is_member(self, assignment: Assignment, user: User) -> bool:
         """Check if a user is a member of an assignment."""
-        return bool(
-            self._db.query(AssignmentMembership)
-            .filter_by(user=user, assignment=assignment)
-            .one_or_none()
-        )
+        return bool(assignment.membership.filter_by(user=user).one_or_none())
```
Comment (marcospri) [inline lms/services/assignment.py:203]: Use the new model field to fetch these.

File: lms/models/assignment.py (hunk 1 above)
Comment (marcospri) [reply lms/models/assignment.py:84]: `dynamic` makes this a query so que can filter by roles etc.

File: lms/services/assignment.py [hunk 3]
```diff
@@ -198,11 +200,19 @@ def get_by_id(self, id_: int) -> Assignment | None:
...
+
+    def get_members(
+        self, assignment, role_type: RoleType, role_scope: RoleScope
+    ) -> list[User]:
+        return [
+            member.user
+            for member in assignment.membership.options(
+                joinedload(AssignmentMembership.user)
```
Comment (marcospri) [inline lms/services/assignment.py:211]: joinedload fetch all users in one go to avoid having to make a query per each one.

**Review by acelaya — APPROVED** [review]

**Pull request merged**